src/Build/Instance/TaskRegistry.cs

```diff
@@ -660,7 +682,22 @@ private void RegisterTask(string taskName, AssemblyLoadInfo assemblyLoadInfo, st
                 _taskRegistrations[taskIdentity] = registeredTaskEntries;
             }
 
-            registeredTaskEntries.Add(new RegisteredTaskRecord(taskName, assemblyLoadInfo, taskFactory, taskFactoryParameters, inlineTaskRecord));
+            RegisteredTaskRecord newRecord = new RegisteredTaskRecord(taskName, assemblyLoadInfo, taskFactory, taskFactoryParameters, inlineTaskRecord);
+
+            if (overrideTask)
+            {
+                if (overriddenTasks.ContainsKey(taskName))
+                {
+                    loggingService.LogWarning(context, null, new BuildEventFileInfo(projectUsingTaskInXml.OverrideLocation), "DuplicateOverrideUsingTaskElement", taskName);
```

Comment:
Why a warning and not an error?

Reply:
https://github.com/dotnet/msbuild/pull/6783#issuecomment-973283563

The conversation has mostly been about whether or not to log a warning, but could also be applied for logging an error. 

IMO it's not worth failing the whole build over. My thinking is the resulting build could still be valid if the user sees this warning. 

> **could** still be valid

Or is that precisely why it's worth failing the whole build over?

Reply:
My thinking is: we can always back it down to a warning if we discover a good reason to. But if we start as a warning we can't go to an error if we discover a good reason that way. So I prefer errors for new scenarios.